### PR TITLE
fix(route): picnob

### LINKED ISSUE #12298: Error when using picnob
### Routes

```routes
/picnob/user/:id
```

### Full routes

```fullroutes
/picnob/user/:id
```

### Related documentation

https://docs.rsshub.app/en/social-media.html#picnob

### What is expected?

Return rss

### What is actually happening?

Error message: Response code 403 (Forbidden): target website might be blocking our access, you can [host your own RSSHub instance](https://docs.rsshub.app/install/) for a better usability.

### Deployment information

Self-hosted

### Deployment information (for self-hosted)

Docker version 20.10.21, build baeda1f

### Additional info

```shell
Helpful Information to provide when opening issue:
Path: /user/fragile.club
Node version: v18.15.0
Git Hash: 64a5a7d
```

### This is not a duplicated issue

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) to ensure this bug has not already been reported

### PR DESCRIPTION
**Pull request by micheal-death**
## Involved Issue / 该 PR 相关 Issue

Close #12298

## Example for the Proposed Route(s) / 路由地址示例

```routes
/picnob/user/xlisa_olivex
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
引入puppeteer，修复了picnob的路由

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
Please use only one browser session to finish all requests instead of having [one request per browser session](https://github.com/DIYgod/RSSHub/actions/runs/7127198827/job/19406618378#step:12:16).

**Review by TonyRL — COMMENTED** [review]
This route still works without puppeteer.

<details><summary>Screenshot</summary>

image
</details>

**Comment by micheal-death** [comment]
> This route still works without puppeteer.
> 
> Screenshot
> image

WX20231208-194207@2x
Wow! How did you make it? I keep getting 403 errors without puppeteer.

**Comment by micheal-death** [comment]
> > > This route still works without puppeteer.
> > > Screenshot
> > 
> > 
> > Wow! How did you make it? I keep getting 403 errors without puppeteer.
> 
> FYI
> 
> Screenshot

Okay, I will try to make another version without puppeteer, thanks!

**Comment by micheal-death** [comment]
@TonyRL I still can't bypass Cloudflare's 403 error on my end, even though I added headers like you mentioned before. Am I missing something?

File: lib/v2/picnob/utils.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const puppeteerGet = async (url, browser) => {
```
Comment (TonyRL) [inline lib/v2/picnob/utils.js:1]: Please block unnecessary requests whe using puppeteer.
docs: https://docs.rsshub.app/joinus/new-rss/start-code#additional-resources

File: lib/v2/picnob/user.js [hunk 2]
```diff
@@ -3,53 +3,90 @@ const cheerio = require('cheerio');
...
+    let html;
+    let usePuppeteer = false;
+    try {
+        const { data } = await got(url, {
+            headers: {
+                accept: 'text/html',
+                referer: 'https://www.google.com/',
+            },
+        });
+        html = data.response;
```
Comment (TonyRL) [inline lib/v2/picnob/user.js:24]:
```suggestion
        html = data;
```

**Pull request merged**